### PR TITLE
gc_worker: Limit the key range to scan for GcKeys tasks

### LINKED ISSUE #11752: GcKeys tasks in GC worker block UnsafeDestroyRange requests and make GC stuck
### What version of TiKV are you using?

/ # /tikv-server -V
TiKV
Release Version:   5.4.0-alpha
Edition:           Community
Git Commit Hash:   28bf633141e05fa9ce1e04444bc7865da258ef02
Git Commit Branch: heads/refs/tags/v5.4.0-nightly
UTC Build Time:    2021-12-30 00:11:41
Rust Version:      rustc 1.56.0-nightly (2faabf579 2021-07-27)
Enable Features:   jemalloc mem-profiling portable sse test-engines-rocksdb cloud-aws cloud-gcp
Profile:           dist_release

### Steps to reproduce

oltp_rel_008, drop one database, inject tikv instance down some times, after tikv recover, check the gc safepoint.

### What did you expect?
gc safepoint should push ahead after recover from tikv failure.

### What did happened?
image

### LINKED ISSUE #11902: GC Keys tasks executes slowly due to millions of seek_tombstone
### What version of TiKV are you using?

/ # /tikv-server -V
TiKV
Release Version:   5.4.0-alpha
Edition:           Community
Git Commit Hash:   e8718be24177a7218c317ade7d8055a3a1c4d6b5
Git Commit Branch: heads/refs/tags/v5.5.0-nightly
UTC Build Time:    2022-01-24 06:09:14
Rust Version:      rustc 1.59.0-nightly (cfa4ac66c 2022-01-06)
Enable Features:   jemalloc mem-profiling portable sse test-engines-rocksdb cloud-aws cloud-gcp cloud-azure
Profile:           dist_release

### What did you expect?
Should be dropped during serval minutes.

### What did happened?
280G took 2 hours to drop.

From GC tasks duration, we can see that gc_keys is very slow. Since total size to be dropped only has 280G, the seek-tombstone in GC scan write detail seems abnormal.   

image

image

image

image

Slow gc_keys tasks may block execution of unsafe_destroy_range tasks and further blocks TiDB's GC procedure (causes #11752)

### PR DESCRIPTION
**Pull request by MyonKeminta**
Signed-off-by: MyonKeminta <MyonKeminta@users.noreply.github.com>

### What is changed and how it works?

Close #11902
Close #11752
What's Changed: Limits the range to scan when processing `GcKeys` tasks in GC Worker. Hopefully this may reduce unnecessary tombstone scan in GcWorker that slows down GC tasks.

### Related changes

- PR to update `pingcap/docs`/`pingcap/docs-cn`:
- PR to update `pingcap/tidb-ansible`:
- Need to cherry-pick to the release branch

### Check List 

Tests 

- Unit test

Side effects

-

### Release note 

```release-note
None
```

### CODE REVIEW

**Review by Connor1996 — APPROVED** [review]
LGTM. Does it relieve the issue?

**Comment by MyonKeminta** [comment]
> LGTM. Does it relieve the issue?

Waiting for test result from @mayjiang0203 .

File: src/server/gc_worker/gc_worker.rs [hunk 1]
```diff
@@ -351,9 +351,21 @@ where
...
 
         let count = keys.len();
+        let range_start_key = keys.first().unwrap().clone().into_encoded();
+        let range_end_key = {
+            let mut k = keys
+                .last()
+                .unwrap()
+                .to_raw()
+                .map_err(|e| EngineError::Codec(e))?;
+            k.push(b'0');
```
Comment (youjiali1995) [inline src/server/gc_worker/gc_worker.rs:361]: `b'0'` == 48
```suggestion
            k.push(0);
```

**Review by hicqu — APPROVED** [review]

**Comment by MyonKeminta** [comment]
Before:
image
image


After:
origin_img_v2_59949277-41f6-403b-9435-0558ee45cdcg
image

**Pull request merged**